mlflow/pyfunc/__init__.py

```diff
@@ -1153,6 +1169,12 @@ def spark_udf(
         BooleanType,
     )
 
+    is_spark_connect_session = _is_spark_connect_session(spark)
+    if is_spark_connect_session and env_manager in (_EnvManager.VIRTUALENV, _EnvManager.CONDA):
+        raise MlflowException.invalid_parameter_value(
+            f"{env_manager!r} environment manager is not supported for Spark Connect",
+        )
```

Comment:
For virtualenv and conda mode, we can support connect mode by setting up python env in spark worker side .

Reply:
how slow is it?

Reply:
In this way, every spark task has to download the python binaries and dependency packages . It is slow but it is a valid fallback approach.

Reply:
We can support it in follow-up PR. 
> every spark task has to download the python binaries and dependency packages

This might be the only way for now, but we can still apply the same optimization similar to https://github.com/mlflow/mlflow/pull/9024#discussion_r1267517598